crypto/s2n_ecc_evp.c

```diff
@@ -143,6 +154,223 @@ int s2n_ecc_evp_compute_shared_secret_from_params(struct s2n_ecc_evp_params *pri
     return 0;
 }
 
+#if (!MODERN_EC_SUPPORTED)
+static int s2n_ecc_evp_calculate_point_length(const EC_POINT *point, const EC_GROUP *group, uint8_t *length) {
+    size_t ret = EC_POINT_point2oct(group, point, POINT_CONVERSION_UNCOMPRESSED, NULL, 0, NULL);
+    S2N_ERROR_IF(ret == 0, S2N_ERR_ECDHE_SERIALIZING);
+    S2N_ERROR_IF(ret > UINT8_MAX, S2N_ERR_ECDHE_SERIALIZING);
+    *length = (uint8_t)ret;
+    return 0;
+}
+
+static int s2n_ecc_evp_write_point_data_snug(const EC_POINT *point, const EC_GROUP *group, struct s2n_blob *out) {
+    size_t ret = EC_POINT_point2oct(group, point, POINT_CONVERSION_UNCOMPRESSED, out->data, out->size, NULL);
+    S2N_ERROR_IF(ret != out->size, S2N_ERR_ECDHE_SERIALIZING);
+    return 0;
+}
+
+static EC_POINT *s2n_ecc_evp_blob_to_point(struct s2n_blob *blob, const EC_KEY *ec_key) {
+    const EC_GROUP *group = EC_KEY_get0_group(ec_key);
+    EC_POINT *point = EC_POINT_new(group);
+    if (point == NULL) {
+        S2N_ERROR_PTR(S2N_ERR_ECDHE_UNSUPPORTED_CURVE);
+    }
+    if (EC_POINT_oct2point(group, point, blob->data, blob->size, NULL) != 1) {
+        EC_POINT_free(point);
+        S2N_ERROR_PTR(S2N_ERR_BAD_MESSAGE);
+    }
+    return point;
+}
+#endif
+
+int s2n_ecc_evp_read_params_point(struct s2n_stuffer *in, int point_size, struct s2n_blob *point_blob) {
+    notnull_check(in);
+    notnull_check(point_blob);
+    gte_check(point_size, 0);
+
+    /* Extract point from stuffer */
+    point_blob->size = point_size;
+    point_blob->data = s2n_stuffer_raw_read(in, point_size);
+    notnull_check(point_blob->data);
+
+    return 0;
+}
+
+int s2n_ecc_evp_read_params(struct s2n_stuffer *in, struct s2n_blob *data_to_verify,
+                            struct s2n_ecdhe_raw_server_params *raw_server_ecc_params) {
+    notnull_check(in);
+    uint8_t curve_type;
+    uint8_t point_length;
+
+    /* Remember where we started reading the data */
+    data_to_verify->data = s2n_stuffer_raw_read(in, 0);
+    notnull_check(data_to_verify->data);
+
+    /* Read the curve */
+    GUARD(s2n_stuffer_read_uint8(in, &curve_type));
+    S2N_ERROR_IF(curve_type != s2n_ecc_evp_supported_curves_list_len, S2N_ERR_BAD_MESSAGE);
```

Comment:
`s2n_ecc_evp_supported_curves_list_len` is probably not what you want to be comparing against here. It might have the correct value, but it almost definitely isn't the correct variable/constant.

Reply:
This a good catch, updating to using constant `#define TLS_EC_CURVE_TYPE_NAMED 3`.